src/storage/cache/rpc/client.rs

```diff
@@ -0,0 +1,640 @@
+                debug!("{:?} Failed to receive response header: {:?}", self, err);
+                Err(RpcError::InternalError(err.to_string()))
+            }
+        }
+    }
+
+    /// Send a request to the server.
+    /// We need to make sure
+    /// Current send packet need to be in one task, so if we need to send ping and packet
+    /// we need to make sure only one operation is running, like select.
+    pub async fn send_data(
+        &self,
+        req_header: &dyn Encode,
+        req_body: Option<&dyn Encode>,
+    ) -> Result<(), RpcError> {
+        let buf = unsafe { &mut *self.req_buf.get() };
+        buf.clear();
+        // encode just need to append to buffer, do not clear buffer
+        req_header.encode(buf);
+        // Append the body to the buffer
+        if let Some(body) = req_body {
+            // encode just need to append to buffer, do not clear buffer
+            body.encode(buf);
+        }
+        debug!("{:?} Sent data with length: {:?}", self, buf.len());
+        let writer = self.get_stream_mut();
+        match write_all_timeout!(writer, buf, self.timeout_options.write_timeout).await {
+            Ok(()) => Ok(()),
+            Err(err) => {
+                debug!("{:?} Failed to send data: {:?}", self, err);
+                Err(RpcError::InternalError(err.to_string()))
+            }
+        }
+    }
+
+    /// Get the next sequence number.
+    pub fn next_seq(&self) -> u64 {
+        self.seq.fetch_add(1, std::sync::atomic::Ordering::AcqRel)
+    }
+
+    /// Send keep alive message to the server
+    pub async fn ping(&self) -> Result<(), RpcError> {
+        // Send keep alive message
+        let current_seq = self.next_seq();
+        let current_timestamp = self.clock_instant.elapsed().as_secs();
+        // Check keepalive is valid
+        let received_keepalive_timestamp = self
+            .received_keepalive_timestamp
+            .load(std::sync::atomic::Ordering::Acquire);
+        if current_timestamp - received_keepalive_timestamp
+            > self.timeout_options.keep_alive_timeout.as_secs()
+        {
+            debug!("{:?} keep alive timeout, close the connection", self);
+            return Err(RpcError::InternalError("Keep alive timeout".to_owned()));
+        }
+
+        // Set to packet task
+        let keep_alive_msg = ReqHeader {
+            seq: current_seq,
+            op: ReqType::KeepAliveRequest.to_u8(),
+            len: 0,
+        };
+
+        if let Ok(()) = self.send_data(&keep_alive_msg, None).await {
+            debug!("{:?} Success to sent keep alive message", self);
+            Ok(())
+        } else {
+            debug!("{:?} Failed to send keep alive message", self);
+            Err(RpcError::InternalError(
+                "Failed to send keep alive message".to_owned(),
+            ))
+        }
+    }
+
+    /// Send packet by the client.
+    pub async fn send_packet(&self, req_packet: &mut P) -> Result<(), RpcError> {
```

Comment:
`req_packet: &mut P `  change to `req_packet: mut P`

Reply:
Ok.

Reply:
It seems that function does not support mut P.

Reply:
OK.

Reply:
Changed to `pub async fn send_packet(&self, mut req_packet: P)`.